sdks/go/pkg/beam/util/gcsx/gcs.go

```diff
@@ -63,6 +64,41 @@ func Upload(ctx context.Context, client *storage.Client, project, bucket, object
 
 }
 
+// SoftDeletePolicyEnabled checks whether soft delete policy is enabled.
+// Note that soft delete policy is disabled when RetentionDuration is zero. Otherwise,
+// soft delete policy is enabled by default.
+func SoftDeletePolicyEnabled(attrs *storage.BucketAttrs) bool {
+	if attrs == nil || attrs.SoftDeletePolicy == nil {
+		return false
+	}
+	return attrs.SoftDeletePolicy.RetentionDuration > 0
+}
+
+// WarnIfSoftDeletePolicyEnabled warns if the soft delete policy is enabled for the temporary bucket.
+func WarnIfSoftDeletePolicyEnabled(ctx context.Context, client *storage.Client, bucket string) {
+	exists, err := BucketExists(ctx, client, bucket)
+	if err != nil {
+		panic("Fail to check BucketExists.")
```

Comment:
It's not a good idea to panic deep into libraries. Consider returning an error instead or simply silently return. 

In this case, it's unexpected to panic if it's called on a non existent bucket, when the function's goal is to log a message. 

Return an error perhaps,but certainly not crash the program.